ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -566,12 +612,12 @@ public FindInfo findEntriesSince(FindToken token, long maxTotalSizeOfEntries) th
       logger.trace("Time used to validate token: {}", (time.milliseconds() - startTimeInMs));
 
       List<MessageInfo> messageEntries = new ArrayList<MessageInfo>();
-      if (storeToken.getStoreKey() == null) {
+      if (!storeToken.getType().equals(StoreFindToken.Type.IndexBased)) {
         startTimeInMs = time.milliseconds();
         boolean inclusive = false;
-        long offsetToStart = storeToken.getOffset();
-        if (offsetToStart == StoreFindToken.Uninitialized_Offset) {
-          offsetToStart = 0;
+        Offset offsetToStart = storeToken.getOffset();
+        if (storeToken.getType().equals(StoreFindToken.Type.Uninitialized)) {
+          offsetToStart = log.getStartOffset();
           inclusive = true;
         } else if (tokenWasReset) {
           offsetToStart = logEndOffsetOnStartup;
```

Comment:
Take a look at the code a few lines above where the value of `tokenWasReset` is set. It is set to `true` if the offset in the token is > `logEndOffsetOnStartup`. So `offsetToStart = logEndOffsetOnStartup` is correct.